### PR TITLE
Upgrades for 2022

### PR DESCRIPTION
**Pull request by virtuald**
Needs testing.

### CODE REVIEW

**Comment by auscompgeek** [comment]
I'd still want to do a bit more testing (mainly on Windows), but this looks fairly good to go to me. I think Dustin was contemplating a couple more changes though.

**Comment by virtuald** [comment]
Thanks for doing the bulk of the work.

I would like to change the package to leverage the existing wpiutil robotpy-build package if it's present in the environment, otherwise fallback to the standalone version we have now. I would also like to test it on a real RoboRIO. Won't have time tonight, maybe tomorrow night.

File: setup.py [hunk 1]
```diff
@@ -138,10 +138,27 @@ def has_flag(compiler, flagname):
...
+    flags = [
+        f"{pfx}std{sep}c++20",
+        f"{pfx}std{sep}c++17",
+    ]
+
+    for flag in flags:
+        if has_flag(compiler, flag):
+            return flag
+
+    raise RuntimeError("Unsupported compiler -- at least C++11 support is needed!")
```
Comment (auscompgeek) [inline setup.py:155]:
```suggestion
    raise RuntimeError("Unsupported compiler -- at least C++17 support is needed!")
```

Comment (virtuald) [reply setup.py:248]: lol oops

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -38,23 +57,35 @@ jobs:
...
+      - run: python3 -m pip install -v dist/*.tar.gz
       - run: python3 -m cscore --help
 
+  cross-build:
+    runs-on: ubuntu-latest
+    container:
+      image: robotpy/roborio-cross-ubuntu:2022.3
+
+    steps:
+    - uses: actions/checkout@v1
```
Comment (auscompgeek) [inline .github/workflows/ci.yml:69]: We should use the sdist from the sdist job here instead.

**Comment by virtuald** [comment]
:shipit:

**Pull request merged**